contracts/TransferLimitModule.sol

```diff
@@ -0,0 +1,445 @@
+    function changeTransferLimit(address token, uint256 transferLimit)
+        public
+        authorized
+    {
+        transferLimits[token].limit = transferLimit;
+    }
+
+    /// @dev Updates the delegate. This can only be done via a Safe transaction.
+    /// @param _delegate New delegate, who is a non-owner account also allowed to make transfers.
+    function setDelegate(address _delegate)
+        public
+        authorized
+    {
+        delegate = _delegate;
+    }
+
+    /// @dev Returns if Safe transaction is a valid transfer limit transaction.
+    /// @param token Address of the token that should be transfered (0 for Ether)
+    /// @param to Address to which the tokens should be transfered
+    /// @param amount Amount of tokens (or Wei) that should be transfered
+    /// @param refund Gas to be refunded to relayer (gasPrice*gasLimit).
+    /// @param gasToken Token address (or 0 if ETH) that is used for the payment.
+    /// @param refundReceiver Address of receiver of gas payment (or 0 if tx.origin).
+    /// @param signatures Packed signature data ({bytes32 r}{bytes32 s}{uint8 v})
+    /// @return Returns if transaction can be executed.
+    function executeTransferLimit(
+        address token,
+        address to,
+        uint256 amount,
+        uint256 refund,
+        address gasToken,
+        address refundReceiver,
+        bytes memory signatures
+    )
+        public
+    {
+        require(to != address(0), "Invalid to address provided");
+        require(amount > 0, "Invalid amount provided");
+
+        bytes32 txHash = getTransactionHash(
+            token, to, amount,
+            refund, gasToken, refundReceiver,
+            nonce
+        );
+        require(checkSignatures(txHash, signatures), "Invalid signatures provided");
+        // Increase nonce and execute transaction.
+        nonce++;
+
+        // If time period is over, reset expenditure.
+        if (isPeriodOver()) {
+            transferLimits[token].spent = 0;
+            totalWeiSpent = 0;
+            totalDaiSpent = 0;
+        }
+
+        // Validate that transfer is not exceeding transfer limit, and
+        // compute new expenditure values to be updated after transfering.
+        (bool isUnderLimit, uint256 weiSpent, uint256 daiSpent) = checkTransferLimits(token, amount);
+        require(isUnderLimit, "Transfer exceeds limits");
+
+        // Perform transfer by invoking manager
+        bool ok;
+        if (token == address(0)) {
+            ok = manager.execTransactionFromModule(to, amount, "", Enum.Operation.Call);
+        } else {
+            bytes memory data = abi.encodeWithSignature("transfer(address,uint256)", to, amount);
+            ok = manager.execTransactionFromModule(token, 0, data, Enum.Operation.Call);
+        }
+
+        // Only update transfer limits if transfer happened.
+        if (ok) {
+            updateTransferLimits(token, amount, weiSpent, daiSpent);
```

Comment:
Does this allow a reentrency attack? Since we perform the transfer that could trigger another transaction (with a valid signature and nonce) to this module and both could use the maximum limit.